modules/oe_whitelabel_modal_page/tests/src/Functional/ModalPageTemplateOverrideTest.php

```diff
@@ -0,0 +1,469 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_whitelabel_modal_page\Functional;
+
+use Drupal\Tests\BrowserTestBase;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests that modal configuration options render correctly through templates.
+ */
+class ModalPageTemplateOverrideTest extends BrowserTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'oe_whitelabel_helper',
+    'oe_whitelabel_modal_page',
+    'filter',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'oe_whitelabel';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Configure modal_page for Bootstrap 5.x to enable toast support.
+    $this->config('modal_page.settings')
+      ->set('modal_provider', 'bootstrap')
+      ->set('bootstrap_version', '5x')
+      ->save();
+
+    // Create a basic HTML text format for testing formatted body content.
+    $format = \Drupal::entityTypeManager()->getStorage('filter_format')->create([
+      'format' => 'basic_html',
+      'name' => 'Basic HTML',
+      'filters' => [
+        'filter_html' => [
+          'status' => 1,
+          'settings' => [
+            'allowed_html' => '<p> <br> <strong> <em> <a>',
+          ],
+        ],
+      ],
+    ]);
+    $format->save();
+
+    // Create a reusable test modal with default configuration.
+    $modal_storage = \Drupal::entityTypeManager()->getStorage('modal');
+    $modal = $modal_storage->create([
+      'id' => 'test_modal',
+      'label' => 'Test modal title',
+      'body' => [
+        'value' => 'Test modal body content.',
+        'format' => 'plain_text',
+      ],
+      'pages' => '/user/login',
+      'auto_open' => TRUE,
+      'published' => TRUE,
+      'type' => 'page',
+    ]);
+    $modal->save();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function tearDown(): void {
+    /** @var \Drupal\modal_page\Entity\ModalInterface $modal */
+    $modal = \Drupal::entityTypeManager()->getStorage('modal')->load('test_modal');
+    if ($modal) {
+      $modal->delete();
```

Comment:
Why do we need this?
I can see why we would need it in an ExistingSite test, but here?